### PR TITLE
add openwrt support

### PR DESCRIPTION
**Pull request by qdk0901**
add cmake file
because mipsel-openwrt-linux-uclibc-gcc-ar in openwrt toolchain will report "sorry - this program has been built without plugin support" error, just replace with mipsel-openwrt-linux-uclibc-ar can fix the error,
so some changes in CMakeLists.txt is needed

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -99,8 +103,13 @@ project (Jerry CXX C ASM)
    get_filename_component(PATH_TO_GCC ${CMAKE_C_COMPILER} REALPATH)
    get_filename_component(DIRECTORY_GCC ${PATH_TO_GCC} DIRECTORY)
    get_filename_component(FILE_NAME_GCC ${PATH_TO_GCC} NAME)
-   string(REPLACE "gcc" "gcc-ar" CMAKE_AR ${FILE_NAME_GCC})
-   string(REPLACE "gcc" "gcc-ranlib" CMAKE_RANLIB ${FILE_NAME_GCC})
+   if("${PLATFORM}" STREQUAL "OPENWRT")
+	string(REPLACE "gcc" "ar" CMAKE_AR ${FILE_NAME_GCC})
```
Comment (LaszloLango) [inline CMakeLists.txt]: Please avoid using of tab characters

**Comment by seanshpark** [comment]
LGTM

File: CMakeLists.txt [hunk 2]
```diff
@@ -99,8 +103,13 @@ project (Jerry CXX C ASM)
...
    get_filename_component(DIRECTORY_GCC ${PATH_TO_GCC} DIRECTORY)
    get_filename_component(FILE_NAME_GCC ${PATH_TO_GCC} NAME)
-   string(REPLACE "gcc" "gcc-ar" CMAKE_AR ${FILE_NAME_GCC})
-   string(REPLACE "gcc" "gcc-ranlib" CMAKE_RANLIB ${FILE_NAME_GCC})
+   if("${PLATFORM}" STREQUAL "OPENWRT")
+    string(REPLACE "gcc" "ar" CMAKE_AR ${FILE_NAME_GCC})
+    string(REPLACE "gcc" "ranlib" CMAKE_RANLIB ${FILE_NAME_GCC})
+   else()
+    string(REPLACE "gcc" "gcc-ar" CMAKE_AR ${FILE_NAME_GCC})
+    string(REPLACE "gcc" "gcc-ranlib" CMAKE_RANLIB ${FILE_NAME_GCC})
```
Comment (ruben-ayrapetyan) [inline CMakeLists.txt]: This is necessary for LTO build, but should be not necessary if LTO is turned off.

Do I understand correctly that LTO is not supported on OpenWRT?
If so, `ENABLE_LTO` should be turned off for OpenWRT builds and this `if` block should become unnecessary.
Could you, please, try building the target without adding the `if`, using `make LTO=OFF ...`?

**Comment by qdk0901** [comment]
@ruben-ayrapetyan 
Hi, sorry, when I pull the latest jerry submodule in iotjs, the change in CMakeList.txt is not needed anymore

**Comment by ruben-ayrapetyan** [comment]
@qdk0901, could you, please, update the pull request to remove the change that is not needed, and combine all commits of the pull request into one with signed-off message?

**Comment by qdk0901** [comment]
@ruben-ayrapetyan sorry， I don't know how to clean the pull request messages here， so I make a new [pull request](https://github.com/Samsung/jerryscript/pull/837)，please close this one

**Pull request closed without merging**